src/framework/analyses.ml

```diff
@@ -268,19 +268,9 @@ struct
       printf "Writing Sarif to file: %s\n%!" (get_string "outfile");
       Yojson.Safe.pretty_to_channel ~std:true out (Sarif.to_yojson (List.rev !Messages.Table.messages_list));
     | "json-messages" ->
-      let files =
-        let module SH = BatHashtbl.Make (Basetype.RawStrings) in
-        let files = SH.create 100 in
-        iterGlobals file (function
-            | GFun (_, loc)
-            | GVar (_, _, loc) ->
-              SH.replace files loc.file (Hashtbl.find_option Preprocessor.dependencies loc.file)
-            | _ -> () (* TODO: add locs from everything else? would also include system headers *)
-          );
-        files |> SH.to_list
-      in
+      let files = Hashtbl.to_list Preprocessor.dependencies in
       let json = `Assoc [
-          ("files", `Assoc (List.map (Tuple2.map2 [%to_yojson: string list option]) files));
+          ("files", `Assoc (List.map (Tuple2.map2 [%to_yojson: (string*bool) list]) files));
```

Comment:
I didn't look at the output, but if I remember correctly, these pairs show up as JSON arrays like `["file.c", false]`, which aren't very good style. I'm thinking maybe we should just filter out the system ones and leave the non-system filenames (without the booleans).
Of course it's true that the preprocessing depended on those system headers, but I don't think there's value in watching them for changes, etc, because we already assume in many other ways that GCC doesn't randomly change (e.g. CIL's `Machdep` would have to be forcefully rebuilt then anyway).